File: TradeHarborApi/Models/Friend/UserProfile.cs
namespace TradeHarborApi.Models
{
    /// <summary>
    /// Represents a user profile in relation to the user with JWT.
    /// </summary>
    public class UserProfile
    {
        /// <summary>
        /// Gets or sets the user ID.
        /// </summary>
        public string UserId { get; set; } = string.Empty;

        /// <summary>
        /// Gets or sets the first name.
        /// </summary>
        public string FirstName { get; set; } = string.Empty;

        /// <summary>
        /// Gets or sets the last name.
        /// </summary>
        public string LastName { get; set; } = string.Empty;

        /// <summary>
        /// Gets or sets the username.
        /// </summary>
        public string Username { get; set; } = string.Empty;

        /// <summary>
        /// Gets or sets the profile picture URL.
        /// </summary>
        public string ProfilePictureUrl { get; set; } = string.Empty;

        /// <summary>
        /// Gets or sets a value indicating whether a friend request was sent to the user.
        /// </summary>
        public bool SentFriendRequestToYou { get; set; } = false;

        /// <summary>
        /// Gets or sets a value indicating whether a friend request was received from the user.
        /// </summary>
        public bool ReceivedFriendRequestFromYou { get; set; } = false;

        /// <summary>
        /// Gets or sets a value indicating whether the user is a friend.
        /// </summary>
        public bool IsFriend { get; set; } = false;
    }
}

File: TradeHarborClient/src/app/models/api-responses/userProfile.ts
/**
 * Represents a user profile with basic information and friendship status.
 */
export class UserProfile {
    /**
     * The unique identifier for the user.
     */
    userId: string = '';
  
    /**
     * The first name of the user.
     */
    firstName: string = '';
  
    /**
     * The last name of the user.
     */
    lastName: string = '';
  
    /**
     * The username of the user.
     */
    username: string = '';
  
    /**
     * The URL of the user's profile picture.
     */
    profilePictureUrl: string = '';
  
    /**
     * Indicates whether the user has sent a friend request to you.
     */
    sentFriendRequestToYou: boolean = false;
  
    /**
     * Indicates whether you have received a friend request from the user.
     */
    receivedFriendRequestFromYou: boolean = false;
  
    /**
     * Indicates whether the user is a friend.
     */
    isFriend: boolean = false;
  }
  